pkg/controller/controllercmd/builder.go

```diff
@@ -188,6 +190,12 @@ func (b *ControllerBuilder) WithEventRecorderOptions(options record.CorrelatorOp
 	return b
 }
 
+// WithComponentOwnerReference overrides controller reference resolution for event recording
+func (b *ControllerBuilder) WithComponentOwnerReference(reference corev1.ObjectReference) *ControllerBuilder {
```

Comment:
nit: *corev1.ObjectReference

Reply:
I wanted to prevent the mutation from outside, but I suppose the implementation will be easier with a pointer